spot-client/src/spot-tv/backend/SpotTvBackendService.js

```diff
@@ -70,6 +71,32 @@ export class SpotTvBackendService extends SpotBackendService {
         return this.remotePairingInfo.code;
     }
 
+    /**
+     * Makes request to the backend and tries to get the exit password. If the requests fails will return the most
+     * recent cached value.
+     *
+     * @returns {Promise<?string>}
+     */
+    fetchExitPassword() {
+        if (!this._getExitPasswordPromise) {
+            this._getExitPasswordPromise = fetchRoomInfo(
+                this.roomKeeperServiceUrl,
+                this.getJwt()
+            ).then(({ endpointPassword }) => {
+                this._cachedExitPassword = endpointPassword && endpointPassword.length ? endpointPassword : undefined;
+            }, error => {
+                logger.error('Fetch exit password failed', { error });
```

Comment:
Do you need to clear _getExitPasswordPromise? Is it really necessary to gate the execution of `fetchExitPassword` with `this. _getExitPasswordPromise`?

Reply:
The intention was to prevent queuing multiple requests, but it looks like the UI ended up not allowing that. I will revert it.

Reply:
The promise was being cleared in the last .then() which is the common code executed for either resolve or reject.